src/users/user.entity.ts

```diff
@@ -35,4 +36,9 @@ export class User {
 
   @VersionColumn()
   dataVersion: number;
+
+  @BeforeInsert()
+  async hashPassword() {
+    this.password = await bcrypt.hash(this.password, 10);
+  }
```

Comment:
This is specifically why we need to instantiate a `User` from `CreateUserDTO` before insertion, but I guess it should have been like this from the start

Reply:
I understand the code but not your explanation...

Reply:
`CreateUserDTO` is not an entity thus entity decorators won't be applied if we only manipulate `CreateUserDTO` upon insertion

Reply:
https://github.com/soundhive/api/pull/52/files#diff-8bf5e7d65bb479beb05b4a3ef1496d48L21-R24

Reply:
Ok I see, I should be better